### PR TITLE
#166816124 Super admins can create users

### PR DESCRIPTION
**Pull request by dharmykoya**
#### What does this PR do?
Allowing super admins to create users.
#### Description of Task to be completed?
- To write Test for all the functions
- To allow superAdmin to create users
- To allow superAdmin change users roleType
- To allow admins to delete users
- To allow admins to update users information
- To setup pre-commit hooks
#### How should this be manually tested?
N/A
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#166816124](https://www.pivotaltracker.com/story/show/166816124)
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

**Review by tunedev — APPROVED** [review]
Good one @dharmykoya, LGTM.

**Comment by fxola** [comment]
LGTM

**Comment by halimahO** [comment]
LGTM

File: src/db/models/user.js [hunk 1]
```diff
@@ -5,18 +5,36 @@ export default (sequelize, DataTypes) => {
...
+          msg: 'Please enter a valid character'
+        }
       }
     },
     userName: {
       type: DataTypes.STRING,
-      allowNull: true
+      allowNull: true,
+      validate: {
+        isAlpha: {
```
Comment (nedemenang) [inline src/db/models/user.js]: make this alphanumberic

File: src/middlewares/auth.middleware.js [hunk 2]
```diff
@@ -117,11 +117,36 @@ export default {
...
+   * @param {Function} next function
+   *
+   * @returns {Response} response object
+   */
+
+  async adminCheck(request, response, next) {
+    const admin =
+      request.user.roleType === 'super_admin' ||
+      request.user.roleType === 'admin';
+    if (!admin) {
```
Comment (nedemenang) [inline src/middlewares/auth.middleware.js]: put the condition in the if bracket.

**Pull request merged**